### PR TITLE
handle the restartPolicy for play kube and generate kube

### PR DESCRIPTION
**Pull request by zhangguanzhang**
Fixes: https://github.com/containers/podman/issues/7656
Signed-off-by: zhangguanzhang <zhangguanzhang@qq.com>

### CODE REVIEW

File: pkg/domain/infra/abi/play.go [hunk 1]
```diff
@@ -299,6 +299,18 @@ func (ic *ContainerEngine) playKubePod(ctx context.Context, podName string, podY
...
+	var ctrRestartPolicy string
+	switch podYAML.Spec.RestartPolicy {
+	case v1.RestartPolicyAlways:
+		ctrRestartPolicy = libpod.RestartPolicyAlways
+	case v1.RestartPolicyOnFailure:
+		ctrRestartPolicy = libpod.RestartPolicyOnFailure
+	case v1.RestartPolicyNever:
+		ctrRestartPolicy = libpod.RestartPolicyNo
+	default: // Default to Always
+		ctrRestartPolicy = libpod.RestartPolicyAlways
```
Comment (mheon) [inline pkg/domain/infra/abi/play.go:311]: Is this the default? I remember the upstream issue mentioning OnFailure was the default for Kube. @haircommander Would you happen to know this?

**Comment by mheon** [comment]
/approve
One question but otherwise LGTM

File: pkg/domain/infra/abi/play.go (hunk 1 above)
Comment (zhangguanzhang) [reply pkg/domain/infra/abi/play.go:311]:
```
[root@k8s-m1 ~]# kubectl explain pod.spec.restartPolicy
KIND:     Pod
VERSION:  v1

FIELD:    restartPolicy <string>

DESCRIPTION:
     Restart policy for all containers within the pod. One of Always, OnFailure,
     Never. Default to Always. More info:
     https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy
```

File: test/e2e/play_kube_test.go [hunk 2]
```diff
@@ -293,13 +296,14 @@ type HostAlias struct {
...
 	p := Pod{
-		Name:        defaultPodName,
-		Hostname:    "",
-		HostAliases: nil,
-		Ctrs:        make([]*Ctr, 0),
-		Volumes:     make([]*Volume, 0),
-		Labels:      make(map[string]string),
-		Annotations: make(map[string]string),
+		Name:          defaultPodName,
+		RestartPolicy: "",
```
Comment (haircommander) [inline test/e2e/play_kube_test.go:300]: we may want the default here to be `never` as that matches the behavior the tests were written to expect

Comment (zhangguanzhang) [reply test/e2e/play_kube_test.go]: got it

**Comment by rhatdan** [comment]
Could you make sure that the restart policy is also written to the yaml file if `podman generate kube` is run?

**Comment by haircommander** [comment]
> Could you make sure that the restart policy is also written to the yaml file if `podman generate kube` is run?

a difficulty here is podman's restartPolicy is container level, k8s is pod level.

**Comment by mheon** [comment]
Restart policy for pods would be a fun RFE, I might get ambitious and code that up at some point.

**Comment by zhangguanzhang** [comment]
> Could you make sure that the restart policy is also written to the yaml file if `podman generate kube` is run?

It had be done

**Comment by zhangguanzhang** [comment]
ci had error, but no reason 😥, @vrothberg @TomSweeneyRedHat  PTAL

**Comment by vrothberg** [comment]
Restarted the job. Thanks for the ping!

**Comment by zhangguanzhang** [comment]
> Restarted the job. Thanks for the ping!

thanks

File: pkg/domain/infra/abi/play.go (hunk 1 above)
Comment (dtitov) [reply pkg/domain/infra/abi/play.go:311]: @mheon, if you are referring to my issue, then I took this info from the [this doc](http://docs.podman.io/en/latest/markdown/podman-generate-systemd.1.html):
```
–restart-policy=policy

Set the systemd restart policy. The restart-policy must be one of: “no”, “on-success”, “on-failure”, “on-abnormal”, “on-watchdog”, “on-abort”, or “always”. The default policy is on-failure.
```

But now I see that it's a wrong doc. In Kubernetes the default restart policy is indeed `Always`.

**Comment by rhatdan** [comment]
Thanks @zhangguanzhang 
/lgtm

**Pull request merged**